agent/php_execute.c

```diff
@@ -1507,3 +1507,67 @@ void nr_php_user_instrumentation_from_opcache(TSRMLS_D) {
 end:
   nr_php_zval_free(&status);
 }
+
+/*
+ * nr_php_execute_observer fcall_begin and nr_php_execute_observer_fcall_end
+ * are Observer API function handlers that should replicate
+ * the functionality of nr_php_execute_enabled, nr_php_execute,
+ * and nr_php_execute_show that are used when hooking in via zend_execute_ex.
+ *
+ * Observer API functionality was added with PHP 8.0.
+ *
+ * The Observer API provide function handlers that trigger on every userland
+ * function begin and end.  The handlers provide all zend_execute_data and the
+ * end handler provides the return value pointer. The previous way to hook into
+ * PHP was via zend_execute_ex which will hook all userland function calls with
+ * significant overhead for doing the call . However, depending on your stack
+ * size settings, it could potentially generate an extremely deep call stack in
+ * PH because  zend_execute_ex limits your stack size to whatever your settings
+ * are. Observer API bypasses the stack overflow issue that you can run into
+ * when intercepting userland calls.  Additionally, with PHP 8.0, JIT
+ * optimizations could optimize out a call to zend_execute_ex and our extension
+ * would not be able to enter set that call which would lead to segfaults and
```

Comment:
I'm not sure if I capture the intention...
```suggestion
 * optimizations could optimize out a call to zend_execute_ex and agent
 * would not be able to enter that call initialized, which would lead to segfaults and
```

Reply:
Please see if this clarification makes more sense:
https://github.com/newrelic/newrelic-php-agent/pull/501/commits/c5186365eb969aeeba92a2e5344d1f51c3009e6e